integration_tests/testsuite/test_custom.py

```diff
@@ -39,19 +57,179 @@ def runTest(self):
+            for specification in json.loads(output):
                 test_num += 1
-                name = test_info.get('name', 'test_{0}'.format(test_num))
-                path = test_info.get('path')
-                if path is None:
-                    logging.warn('Test \'%s\' has no path specified! '
-                                 'Skipping...', name)
-                    continue
+                self._run_test_for_specification(specification, test_num)
+
+    def _run_test_for_specification(self, specification, test_num):
+        """Given the specification for a test execute the steps and
+            validate the result.
+
+        Args:
+            specification: Dictionary containing the specification.
+            test_num: Identifier of the test.
+
+        Returns:
+            None.
+        """
+        name = specification.get('name', 'test_{0}'.format(test_num))
+        timeout = specification.get('timeout', 3000)
+        steps = specification.get('steps', [])
+        validation = specification.get('validation')
+
+        logging.info('Running custom test: %s', name)
+
+        if self._test_for_old_specification(specification):
+            return
+
+        if validation is None:
+            self.fail('A validation must be specified in the step '
+                      'configuration')
+
+        context = {'name': name}
+        step_num = 0
+
+        for step in steps:
+            step_name, step_context = self._run_step(context, step,
+                                                     step_num, timeout)
+            context[step_name] = step_context
+
+        logging.debug('context : %s', json.dumps(context,
+                                                 sort_keys=True,
+                                                 indent=4,
+                                                 separators=(',', ': ')))
+
+        self._validate(context, validation)
+
+    def _run_step(self, context, step, step_num, timeout):
+        """Use the provided step's configuration to send a request to the
+           specified path and store the result into the context.
+
+        Args:
+            context: A dictionary containing the context for the test.
+            step: A dictionary containing the configuration of the step,
+               this include:
+                 name (optional): name of the step.
+                 configuration (optional):
+                    method: 'GET' or 'POST'.
+                    headers: Dictionary containing the headers of the request.
+                    content: Payload attached to the request.
+                 path: Url of the request
+            step_num: Index of the step.
+
+        Returns:
+            None.
+        """
+        step_name = step.get('name', 'step_{0}'.format(step_num))
+        configuration = step.get('configuration', dict())
+        path = step.get('path')
+
+        logging.info('Running step {0} of test {1}'.format(
```

Comment:
I think it would be useful to include the `path` here as well, not just the name, as path is mandatory anyway.